odxtools/configrecord.py

```diff
@@ -0,0 +1,146 @@
+                "DIAG-COMM-DATA-CONNECTORS/DIAG-COMM-DATA-CONNECTOR")
+        ]
+        config_id = None
+        if (cid_elem := et_element.find("CONFIG-ID")) is not None:
+            config_id = IdentValue.from_et(cid_elem, context)
+        data_records = NamedItemList([
+            DataRecord.from_et(dr_elem, context)
+            for dr_elem in et_element.iterfind("DATA-RECORDS/DATA-RECORD")
+        ])
+        audience = None
+        if (aud_elem := et_element.find("AUDIENCE")) is not None:
+            audience = Audience.from_et(aud_elem, context)
+        system_items = NamedItemList([
+            SystemItem.from_et(si_elem, context)
+            for si_elem in et_element.iterfind("SYSTEM-ITEMS/SYSTEM-ITEM")
+        ])
+        data_id_item = None
+        if (dii_elem := et_element.find("DATA-ID-ITEM")) is not None:
+            data_id_item = DataIdItem.from_et(dii_elem, context)
+        option_items = NamedItemList([
+            OptionItem.from_et(si_elem, context)
+            for si_elem in et_element.iterfind("OPTION-ITEMS/OPTION-ITEM")
+        ])
+        default_data_record_snref = None
+        if (default_data_record_snref_elem :=
+                et_element.find("DEFAULT-DATA-RECORD-SNREF")) is not None:
+            default_data_record_snref = odxrequire(
+                default_data_record_snref_elem.attrib.get("SHORT-NAME"))
+        sdgs = [SpecialDataGroup.from_et(sdge, context) for sdge in et_element.iterfind("SDGS/SDG")]
+
+        return ConfigRecord(
+            config_id_item=config_id_item,
+            diag_comm_data_connectors=diag_comm_data_connectors,
+            config_id=config_id,
+            data_records=data_records,
+            audience=audience,
+            system_items=system_items,
+            data_id_item=data_id_item,
+            option_items=option_items,
+            default_data_record_snref=default_data_record_snref,
+            sdgs=sdgs,
+            **kwargs)
+
+    def _build_odxlinks(self) -> dict[OdxLinkId, Any]:
+        result = {}
+
+        if self.config_id_item is not None:
+            result.update(self.config_id_item._build_odxlinks())
+        for diag_comm_data_connector in self.diag_comm_data_connectors:
+            result.update(diag_comm_data_connector._build_odxlinks())
+        for data_record in self.data_records:
+            result.update(data_record._build_odxlinks())
+        if self.audience is not None:
+            result.update(self.audience._build_odxlinks())
+        for system_item in self.system_items:
+            result.update(system_item._build_odxlinks())
+        if self.data_id_item is not None:
+            result.update(self.data_id_item._build_odxlinks())
+        for option_item in self.option_items:
+            result.update(option_item._build_odxlinks())
+        for sdg in self.sdgs:
+            result.update(sdg._build_odxlinks())
+
+        return result
```

Comment:
A bit off topic, but maybe `_build_odxlinks`, `_resolve_odxlinks` and `_resolve_snrefs`, can be simplified but have one function called `_get_referenceable(self) -> list[Optional[LinkableElement]]`, that return all the objects that should have `_build_odxlinks`, `_resolve_odxlinks` and `_resolve_snrefs` called on them, the rest of the logic can be implemented in the base class.

Reply:
I guess the "leafs" of `_build_odxlinks()` should be moved into `IdentifiableElement` (#425). For resolving the references, I don't really see a good alternative to the current solution. Maybe we can introduce a proxy class like [weakref.proxy](https://docs.python.org/3/library/weakref.html) but I'm not sure if this such a terribly good idea as it is quite "magical"...